server/routerlicious/packages/services-client/src/storageUtils.ts

```diff
@@ -214,3 +216,57 @@ export function convertWholeFlatSummaryToSnapshotTreeAndBlobs(
 		sequenceNumber,
 	};
 }
+
+/**
+ * Converts existing IWholeSummaryTree to ISummaryTree for the first summary (without Handle entries)
+ * @param wholeSummaryTree - wholeSummaryTree used on the payload for creating and uploading a document.
+ * @returns Summary tree to be used when creating a new document.
+ */
+export function convertFirstSummaryWholeSummaryTreeToSummaryTree(
+	wholeSummaryTree: IWholeSummaryTree,
+	unreferenced?: true | undefined,
+): ISummaryTree {
+	const tree: { [path: string]: SummaryObject } = {};
+	for (const entry of wholeSummaryTree.entries) {
+		switch (entry.type) {
+			case "blob": {
+				const blobPayload = (entry as IWholeSummaryTreeValueEntry)
+					.value as IWholeSummaryBlob;
+				tree[entry.path] = {
+					type: SummaryType.Blob,
+					content:
+						blobPayload.encoding === "base64"
+							? new Uint8Array(stringToBuffer(blobPayload.content, "base64"))
+							: blobPayload.content,
+				};
+				break;
+			}
+			case "tree": {
+				const treePayload = (entry as IWholeSummaryTreeValueEntry)
+					.value as IWholeSummaryTree;
+				const nodeReferenced = (entry as IWholeSummaryTreeValueEntry).unreferenced;
```

Comment:
I think instead of using as operator to typecast, you can assert that it is type IWholeSummaryTreeValueEntry and same above. You can use "is" operator.